src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt

```diff
@@ -0,0 +1,109 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.courtsentencing
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.courtsentencing.model.CreateCourtCaseResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.valuesAsStrings
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.SynchronisationMessageType
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.CourtCaseMappingDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.CourtCaseResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.InternalMessage
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationType
+
+@Service
+class CourtSentencingSynchronisationService(
+  private val mappingApiService: CourtSentencingMappingApiService,
+  private val nomisApiService: CourtSentencingNomisApiService,
+  private val dpsApiService: CourtSentencingDpsApiService,
+  private val queueService: SynchronisationQueueService,
+  private val telemetryClient: TelemetryClient,
+) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  suspend fun nomisCourtCaseInserted(event: CourtCaseInsertedEvent) {
+    val telemetry =
+      mapOf(
+        "nomisCourtCaseId" to event.courtCaseId,
+        "offenderNo" to event.offenderIdDisplay,
+        "nomisBookingId" to event.bookingId,
+      )
+    val nomisCourtCase =
```

Comment:
I guess this can be done in the `else` after the module check?

Reply:
yes - revisiting next after the main framework commit.